File: app/src/main/java/funnybrain/hsu/idv/taipeipark/ParkContract.java
package funnybrain.hsu.idv.taipeipark;

import java.util.List;

import funnybrain.hsu.idv.taipeipark.restful.model.Park;

public interface ParkContract {
    interface View extends BaseView<Presenter> {
        void showProgress(boolean show);

        void showPark(List<Park> parks);
    }

    interface Presenter extends BasePresenter {
        void loadTask(boolean forceUpdate);
    }
}

File: app/src/main/java/funnybrain/hsu/idv/taipeipark/ParkContract.kt
package funnybrain.hsu.idv.taipeipark

import funnybrain.hsu.idv.taipeipark.restful.model.Park

interface ParkContract {
    interface View : BaseView<Presenter> {
        fun showProgress(show: Boolean)

        fun showPark(parks: List<Park>)
    }

    interface Presenter : BasePresenter {
        fun loadTask(forceUpdate: Boolean)
    }
}
